virtcontainers/qemu.go

```diff
@@ -653,9 +654,24 @@ func (q *qemu) stopSandbox() error {
 		return err
 	}
 
-	err = os.RemoveAll(filepath.Join(RunVMStoragePath, q.id))
+	// cleanup vm path
+	dir := filepath.Join(RunVMStoragePath, q.id)
+
+	// If it's a symlink, remove both dir and the target.
+	// This can happen when vm template links a sandbox to a vm.
+	link, err := filepath.EvalSymlinks(dir)
 	if err != nil {
-		q.Logger().WithError(err).Error("Fail to clean up vm directory")
+		q.Logger().WithError(err).Warnf("failed to resolve vm path %s", dir)
```

Comment:
I'm presuming this is a non-fatal failure on purpose, so we continue to try and clean up even if we think we cannot find the dir? Maybe we should add an explicit comment for that.